wallet/pskt/src/wasm/pskt.rs

```diff
@@ -384,4 +396,317 @@ impl PSKT {
+                                key_source: None,
+                            })
+                        })
+                    })
+                })
+                .collect::<Result<Vec<_>>>()?;
+
+            Ok(signatures)
+        })?;
+
+        self.replace(State::Signer(signed_pskt))
+    }
+
+    /// Get `Transaction` from a PSKT, by first finalizing it for P2PK.
+    /// This method is useful to broadcast a P2PK PSKT to the Kaspa Network, using `RpcClient.submitTransaction`.
+    #[wasm_bindgen(js_name = "finalizeP2PK")]
+    pub fn finalize_p2pk(&self, network_type: &NetworkTypeT) -> Result<Transaction> {
+        let network_type: NetworkType = network_type.try_into()?;
+
+        let pskt_finalizer: Native<Finalizer> = match self.take() {
+            State::NoOp(inner) => inner.ok_or(Error::NotInitialized)?.into(),
+            State::Creator(pskt) => pskt.constructor().signer().finalizer(),
+            State::Constructor(pskt) => pskt.signer().finalizer(),
+            State::Updater(pskt) => pskt.signer().finalizer(),
+            State::Signer(pskt) => pskt.finalizer(),
+            State::Combiner(pskt) => pskt.signer().finalizer(),
+            State::Finalizer(pskt) => pskt,
+            state => return Err(Error::state(state))?,
+        };
+
+        // pre-conditions check
+        for input in pskt_finalizer.inputs.iter() {
+            if input.redeem_script.is_some() {
+                return Err(Error::custom("finalize_p2pk does not support redeem scripts"));
+            }
+
+            if let Some(utxo_entry) = &input.utxo_entry {
+                let script_class = ScriptClass::from_script(&utxo_entry.script_public_key);
+
+                // only allow p2pk script class
+                match script_class {
+                    ScriptClass::NonStandard => return Err(Error::InvalidScriptClassError(ScriptClass::PubKey, script_class)),
+                    ScriptClass::PubKeyECDSA => return Err(Error::InvalidScriptClassError(ScriptClass::PubKey, script_class)),
+                    ScriptClass::ScriptHash => return Err(Error::InvalidScriptClassError(ScriptClass::PubKey, script_class)),
+                    ScriptClass::PubKey => (),
+                }
+            } else {
+                return Err(Error::custom("Input without UTXO entry cannot be finalized"));
+            }
+        }
+
+        let result = pskt_finalizer.finalize_sync(|inner: &Inner| -> std::result::Result<Vec<Vec<u8>>, String> {
+            inner
+                .inputs
+                .iter()
+                .map(|input| {
+                    if input.partial_sigs.len() != 1 {
+                        return Err(format!(
+                            "finalize_p2pk error: input for outpoint {} must have exactly one signature, but has {}",
+                            input.previous_outpoint,
+                            input.partial_sigs.len()
+                        ));
+                    }
```

Comment:
im not sure it must be error. you can extract what sig you need and leave others

Reply:
I didn't consider multi sig for this implementation indeed, is it what you meant? That there are cases where we could have more than 1 signatue?